txnsync/outgoing_test.go

```diff
@@ -0,0 +1,294 @@
+		requestedTransactionsModulator:     0,
+		requestedTransactionsOffset:        0,
+		lastSentMessageSequenceNumber:      0,
+		lastSentMessageRound:               0,
+		lastSentMessageTimestamp:           0,
+		lastSentMessageSize:                0,
+		lastSentBloomFilter:                bloomFilter{},
+		lastConfirmedMessageSeqReceived:    0,
+		lastReceivedMessageLocalRound:      0,
+		lastReceivedMessageTimestamp:       0,
+		lastReceivedMessageSize:            0,
+		lastReceivedMessageNextMsgMinDelay: 0,
+		dataExchangeRate:                   0,
+		localTransactionsModulator:         0,
+		localTransactionsBaseOffset:        0,
+		lastTransactionSelectionTracker:    nil,
+		nextStateTimestamp:                 0,
+		messageSeriesPendingTransactions:   nil,
+		transactionPoolAckCh:               nil,
+		transactionPoolAckMessages:         nil,
+		lastSelectedTransactionsCount:      0,
+	}
+
+	pendingTransactions := pendingTransactionGroupsSnapshot{
+		pendingTransactionsGroups:           nil,
+		latestLocallyOriginatedGroupCounter: 0,
+	}
+
+	peer.setLocalRequestParams(111, 222)
+	peer.lastReceivedMessageTimestamp = 100
+	peer.lastReceivedMessageLocalRound = s.round
+
+	s.isRelay = true
+
+	metaMessage := s.assemblePeerMessage(&peer, &pendingTransactions)
+
+	a.Equal(metaMessage.message.UpdatedRequestParams.Modulator, byte(222))
+	a.Equal(metaMessage.message.UpdatedRequestParams.Offset, byte(111))
+	a.Equal(metaMessage.peer, &peer)
+	a.Equal(metaMessage.message.Version, int32(txnBlockMessageVersion))
+	a.Equal(metaMessage.message.Round, s.round)
+	a.True(metaMessage.message.MsgSync.ResponseElapsedTime != 0)
+
+}
+
+func TestLocallyGeneratedTransactions(t *testing.T) {
+
+	a := require.New(t)
+
+	pendingTransactions := &pendingTransactionGroupsSnapshot{
+		pendingTransactionsGroups:           []transactions.SignedTxGroup{},
+		latestLocallyOriginatedGroupCounter: 0,
+	}
+
+	s := syncState{}
+
+	pendingTransactions.latestLocallyOriginatedGroupCounter = 1
+
+	a.Equal(s.locallyGeneratedTransactions(pendingTransactions), []transactions.SignedTxGroup{})
+
+	pendingTransactions.pendingTransactionsGroups = []transactions.SignedTxGroup{
+		transactions.SignedTxGroup{
+			Transactions:       nil,
+			LocallyOriginated:  true,
+			GroupCounter:       0,
+			GroupTransactionID: transactions.Txid{},
+			EncodedLength:      2,
+		},
+		transactions.SignedTxGroup{
+			Transactions:       nil,
+			LocallyOriginated:  false,
+			GroupCounter:       0,
+			GroupTransactionID: transactions.Txid{},
+			EncodedLength:      1,
+		},
+		transactions.SignedTxGroup{
+			Transactions:       nil,
+			LocallyOriginated:  true,
+			GroupCounter:       0,
+			GroupTransactionID: transactions.Txid{},
+			EncodedLength:      3,
+		},
+	}
+
+	pendingTransactions.latestLocallyOriginatedGroupCounter = transactions.InvalidSignedTxGroupCounter
+
+	a.Equal(s.locallyGeneratedTransactions(pendingTransactions), []transactions.SignedTxGroup{})
```

Comment:
This case is covering the same code path as the previous case on line 241.
Please add comments to distinguish the covered cases.

Reply:
It does a little bit but I feel that it is ok to cover it again in a dedicated test.

Reply:
I have no issues with covering it again. Actually it is covering for two different reasons which is expected.
I just asked for a comment to describe each case being tested for.